src/openforms/forms/models/form_step.py

```diff
@@ -79,5 +79,11 @@ def __str__(self):
         else:
             return super().__str__()
 
+    def delete(self, using=None, keep_parents=False):
+        super().delete(using=using, keep_parents=keep_parents)
+
+        if self.form_definition.pk is not None and not self.form_definition.is_reusable:
```

Comment:
Why the check on `self.form_definition.pk`?

Reply:
I noticed that in a test(https://github.com/open-formulieren/open-forms/blob/0d58b3a333cfb217f4da0e824aab2257c272f1f1/src/openforms/forms/tests/test_api_import_export.py#L159), we were deleting a formdefinition before the formstep itself. Without this verification, it would raises an error since the form_definition doesn't exist anymore. Nonetheless, I suspect that it wouldn't be possible to replicate this behavior in a real scenario